src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -630,6 +622,8 @@ - (void)updateWithDiscriminator:(const chip::SetupDiscriminator &)deviceDiscrimi
         }
     }
 
+    [self removePeripheralsFromCache];
```

Comment:
This nulls out `_cachedPeripherals`.  Is that expected?

Reply:
No that was not! Thanks !